UnityProject/Assets/Scripts/Items/Weapons/Meleeable.cs

```diff
@@ -1,144 +1,92 @@
 ﻿using System.Collections.Generic;
-using HealthV2;
 using UnityEngine;
 using Items;
 
-//Do not derive from NetworkBehaviour, this is also used on tilemap layers
-/// <summary>
-/// Allows an object or tiles to be attacked by melee.
-/// </summary>
-public class Meleeable : MonoBehaviour, IPredictedCheckedInteractable<PositionalHandApply>
+namespace Systems.Interaction
 {
-	[SerializeField] private bool isMeleeable = true;
-
-	[SerializeField] private static readonly StandardProgressActionConfig ProgressConfig
-		= new StandardProgressActionConfig(StandardProgressActionType.Restrain);
-
-	[SerializeField] private float butcherTime = 2.0f;
-
-	[SerializeField] private string butcherSound = "BladeSlice";
-
+	// Do not derive from NetworkBehaviour, this is also used on tilemap layers
 	/// <summary>
-	/// Which layers are allowed to be attacked on tiles regardless of intent
+	/// Allows an object or tiles to be attacked by melee.
 	/// </summary>
-	private static readonly HashSet<LayerType> attackableLayers = new HashSet<LayerType>(
-		new[]
-		{
+	public class Meleeable : MonoBehaviour, IPredictedCheckedInteractable<PositionalHandApply>
+	{
+		[SerializeField]
+		private bool isMeleeable = true;
+		// If it has this component, isn't it assumed to be meleeable? Is this still true for tilemaps?
```

Comment:
yeah what is this field?

Reply:
No idea. It is only read in one place, though I don't know how many objects are currently using it (or tiles).

Reply:
Rider should be able to get usages of the serialized field. This is weird, like I thought the jobType enum was bad but this is another level of bad lol

Reply:
ah this, it is used to prevent melee attacks on items and objects

Reply:
I've removed the bad dependency of `Integrity` requiring the `Meleeable` component. Instead, new / modified objects should just not attach the `Meleeable` component if it isn't meleeable as opposed to setting the `isMeleeable` property on the component. Though the field is still available for compatibility with current objects.